### PR TITLE
Delta Events

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: packages/node_modules/@ciscospark/internal-plugin-locus/src/locus.js [hunk 1]
```diff
@@ -397,6 +425,74 @@ const Locus = SparkPlugin.extend({
...
+          toRemove.add(p.url);
+        }
+        else {
+          // Elements of the delta events "participants" list that are absent
+          // from the local working copy should be added to that collection.
+          toUpsert.set(p.url, p);
+        }
+      });
+
+      // The "participants" collection in the delta event should be merged with
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/internal-plugin-locus/src/locus.js]: Thanks for the detailed notes in this one!

File: packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+      it(`proceeds through a series of events`, () => handleErrorEvent(spock.spark.phone.dial(room.id), (call) => {
+        const onMembershipConnected = sinon.spy();
+        call.on(`membership:connected`, onMembershipConnected);
+
+        const spockDelta = sinon.spy();
+        const mccoyDelta = sinon.spy();
+
+        spock.spark.internal.mercury.on(`event:locus.difference`, spockDelta);
+        // TODO do not commit the next line; it'll break once delta becomes
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js]: Um?

File: packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                mc.answer()
+              ])
+              .then(() => mc.toggleSendingAudio())
+              .then(() => mc.hangup())
+              ;
+            })
+        ])
+          .then(() => {
+            assert.called(spockDelta);
+            // TODO do not commit the next line
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js]: Another one

File: packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js (hunk 2 above)
Comment (ianwremmel) [reply packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js]: good catch, thanks.

File: packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js (hunk 3 above)
Comment (ianwremmel) [reply packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/delta.js]: good catch, thanks.

File: packages/node_modules/@ciscospark/internal-plugin-locus/src/locus.js (hunk 1 above)
Comment (ianwremmel) [reply packages/node_modules/@ciscospark/internal-plugin-locus/src/locus.js]: haha, i pretty much copied them out of the spec. I'm not entirely sure i understand how this works :)

**Pull request merged**